tiledb/sm/query/writers/domain_buffer.h

```diff
@@ -236,6 +237,11 @@ class DomainBuffersView : public detail::DomainBuffersTypes {
    */
   storage_type qb_;
 
+  /**
+   * Domain.
+   */
+  const Domain& domain_;
```

Comment:
Do we actually need to store this? I only see the domain used inside of the new constructor you added.